packages/frontend-2/components/header/NavShare.vue

```diff
@@ -0,0 +1,133 @@
+            ]"
+            @click="handleCopyLink"
+            @keypress="keyboardClick(handleCopyLink)"
+          >
+            <LinkIcon class="w-5 h-5" />
+            Copy Link
+          </div>
+        </MenuItem>
+        <MenuItem v-if="!isFederated" v-slot="{ active }">
+          <div
+            :class="[
+              active ? 'bg-foundation-focus' : '',
+              'flex gap-2 items-center px-2 py-1.5 text-sm text-foreground cursor-pointer transition rounded'
+            ]"
+            @click="handleCopyId"
+            @keypress="keyboardClick(handleCopyId)"
+          >
+            <FingerPrintIcon class="w-5 h-5" />
+            Copy ID
+          </div>
+        </MenuItem>
+        <MenuItem v-slot="{ active }">
+          <div
+            :class="[
+              active ? 'bg-foundation-focus' : '',
+              'flex gap-2 items-center px-2 py-1.5 text-sm text-foreground cursor-pointer transition rounded'
+            ]"
+            @click="handleEmbed"
+            @keypress="keyboardClick(handleEmbed)"
+          >
+            <CodeBracketIcon class="w-5 h-5" />
+            Embed Model
+          </div>
+        </MenuItem>
+      </MenuItems>
+    </Transition>
+    <ProjectModelPageDialogEmbed
+      v-model:open="embedDialogOpen"
+      :project-id="projectId"
+      :visibility="visibility"
+    />
+  </Menu>
+</template>
+<script setup lang="ts">
+import { Menu, MenuButton, MenuItems, MenuItem } from '@headlessui/vue'
+import {
+  ChevronDownIcon,
+  LinkIcon,
+  FingerPrintIcon,
+  CodeBracketIcon,
+  ShareIcon
+} from '@heroicons/vue/24/outline'
+import { SpeckleViewer } from '@speckle/shared'
+import { keyboardClick } from '@speckle/ui-components'
+import type { ProjectVisibility } from '~/lib/common/generated/gql/graphql'
+import { useCopyModelLink } from '~~/lib/projects/composables/modelManagement'
+
+const props = defineProps<{
+  projectId: string
+  resourceIdString: string
+  visibility: ProjectVisibility
+}>()
+
+const { copy } = useClipboard()
+const copyModelLink = useCopyModelLink()
+
+const embedDialogOpen = ref(false)
+
+const parsedResourceIds = computed(() =>
+  SpeckleViewer.ViewerRoute.parseUrlParameters(props.resourceIdString)
+)
+
+const firstResource = computed(() => parsedResourceIds.value[0] || {})
+
+const versionId = computed(() => {
+  if (SpeckleViewer.ViewerRoute.isModelResource(firstResource.value)) {
+    return firstResource.value.versionId
+  }
+  return ''
+})
+
+const modelId = computed(() => {
+  if (SpeckleViewer.ViewerRoute.isModelResource(firstResource.value)) {
+    return firstResource.value.modelId // Assuming your firstResource object has a modelId property
+  }
+  return ''
+})
+
+const isFederated = computed(() => parsedResourceIds.value.length > 1)
+
+const handleCopyId = () => {
+  copy(props.resourceIdString, { successMessage: 'ID copied to clipboard' })
```

Comment:
In practice this can be multiple IDs, if there are multiple models loaded in the viewer. I'm still not sure if this won't be confusing to the user, tho, cause the only place where we accept IDs with this format is in the viewer URLs. And maybe people get confused if there are multiple comma delimited IDs?

I dunno, I would probably discuss this feature with Benjamin more thoroughly to better make sure this is actually something valuable to our users

Reply:
There is a v-if on the "Copy ID" button, where it won't be shown if there are federated models. handleCopyId can only be called when there is a single model.

Reply:
<img width="453" alt="image" src="https://github.com/specklesystems/speckle-server/assets/139135120/9877d536-8a8f-45dc-9a91-8b5fa67bc00c">